cdap-security/src/main/java/co/cask/cdap/security/store/secretmanager/SecretManagerSecureStoreService.java

```diff
@@ -0,0 +1,200 @@
+  private final SecretManagerContext context;
+  private final String type;
+  private SecretManager secretManager;
+
+  @Inject
+  public SecretManagerSecureStoreService(CConfiguration cConf, NamespaceQueryAdmin namespaceQueryAdmin) {
+    this.namespaceQueryAdmin = namespaceQueryAdmin;
+    this.context = new DefaultSecretManagerContext(cConf);
+    this.type = cConf.get(Constants.Security.Store.PROVIDER);
+    this.secretManager = new SecretManagerExtensionLoader(cConf.get(Constants.Security.Store.EXTENSIONS_DIR)).get(type);
+  }
+
+  @VisibleForTesting
+  public SecretManagerSecureStoreService(NamespaceQueryAdmin namespaceQueryAdmin, SecretManagerContext context,
+                                         String type, SecretManager secretManager) {
+    this.namespaceQueryAdmin = namespaceQueryAdmin;
+    this.context = context;
+    this.type = type;
+    this.secretManager = secretManager;
+  }
+
+  @Override
+  protected void startUp() throws Exception {
+    LOG.info("Starting {}.", getClass().getSimpleName());
+    initializeSecretManager();
+  }
+
+  @Override
+  protected void shutDown() throws Exception {
+    LOG.info("Stopping {}.", getClass().getSimpleName());
+    destroySecretManager();
+  }
+
+  /**
+   * List of all the secrets in the secret manager belonging to the specified namespace.
+   * @return A map of secret name to its description
+   * @param namespace the namespace for list of secrets
+   * @throws NamespaceNotFoundException if the specified namespace does not exist
+   * @throws IOException if there was a problem while reading from the secret manager
+   */
+  @Override
+  public Map<String, String> listSecureData(String namespace) throws Exception {
+    validate(namespace);
+    Map<String, String> map = new HashMap<>();
+    for (SecretMetadata metadata : secretManager.list(namespace)) {
+      map.put(metadata.getName(), metadata.getDescription());
+    }
+
+    return map;
+  }
+
+  /**
+   * Returns the data for the secret identified by the given name.
+   * @param name name of the secret
+   * @return an secure store data identified by the given name
+   * @throws SecureKeyNotFoundException if the secret was not found in the store
+   * @throws IOException if there was a problem in getting the secret from the secret manager
+   */
+  @Override
+  public SecureStoreData getSecureData(String namespace, String name) throws Exception {
+    validate(namespace);
+    try {
+      Secret secret = secretManager.get(namespace, name);
+      SecretMetadata metadata = secret.getMetadata();
+      return new SecureStoreData(new SecureStoreMetadata(metadata.getName(), metadata.getDescription(),
+                                                         metadata.getCreationTimeMs(), metadata.getProperties()),
+                                 secret.getData());
+    } catch (SecretNotFoundException e) {
+      throw new SecureKeyNotFoundException(new SecureKeyId(namespace, name));
```

Comment:
You should preserve the original cause in the new exception, otherwise the stacktrace will be lost.